### PR TITLE
[Improve][Connector-V2] Support read archive compress file

### PR DESCRIPTION
**Pull request by corgy-w**
### Purpose of this pull request
#7587

Temporary support Support for compressed files when reading
zip、tar

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/source/SftpFile.md [hunk 1]
```diff
@@ -176,6 +195,13 @@ The compress codec of files and the details that supported as the following show
 - orc/parquet:  
   automatically recognizes the compression type, no additional settings required.
 
+### archive_compress_codec [string]
+
+The compress codec of archive files and the details that supported as the following shown:
+
+- zip: text,json,xml
+- tar: text,json,xml
+
```
Comment (wuchunfu) [inline docs/en/connector-v2/source/SftpFile.md]: @corgy-w Can tar.gz and more common compression types be supported?

Comment (corgy-w) [reply docs/en/connector-v2/source/SftpFile.md]: get

Comment (corgy-w) [reply docs/en/connector-v2/source/SftpFile.md]: zip, tar, and tar.gz are temporarily supported

File: docs/en/connector-v2/source/CosFile.md [hunk 2]
```diff
@@ -246,7 +265,8 @@ default `HH:mm:ss`
 
 ### schema [config]
 
-Only need to be configured when the file_format_type are text, json, excel, xml or csv ( Or other format we can't read the schema from metadata).
+Only need to be configured when the file_format_type are text, json, excel, xml or csv ( Or other format we can't read
+the schema from metadata).
```
Comment (Hisoka-X) [inline docs/en/connector-v2/source/CosFile.md]: why?

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java [hunk 3]
```diff
@@ -38,6 +38,8 @@
 public class BaseFileSinkConfig implements DelimiterConfig, Serializable {
     private static final long serialVersionUID = 1L;
     protected CompressFormat compressFormat = BaseSinkConfig.COMPRESS_CODEC.defaultValue();
+    protected ArchiveCompressFormat archiveCompressFormat =
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: does sink support this feature?

Comment (corgy-w) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: No support yet， labeled todo in `baseSinkConfig`

Comment (Hisoka-X) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: Please remove it if useless at now.

Comment (corgy-w) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: get

Comment (corgy-w) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseFileSinkConfig.java]: Sink config is deleted. only todo is retained

File: seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseSinkConfig.java [hunk 4]
```diff
@@ -54,6 +54,13 @@ public class BaseSinkConfig extends KerberosConfig {
                     .defaultValue(CompressFormat.NONE)
                     .withDescription("Compression codec");
 
+    // TODO：Compression is supported during write
+    public static final Option<ArchiveCompressFormat> ARCHIVE_COMPRESS_CODEC =
```
Comment (Hisoka-X) [inline seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseSinkConfig.java:58]: Please update https://github.com/apache/seatunnel/blob/4f812e12aeb20c301bdc5bc5fc93b47b6b2cee50/seatunnel-connectors-v2/connector-file/connector-file-sftp/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/sftp/source/SftpFileSourceFactory.java#L43

Comment (corgy-w) [reply seatunnel-connectors-v2/connector-file/connector-file-base/src/main/java/org/apache/seatunnel/connectors/seatunnel/file/config/BaseSinkConfig.java:58]: get

**Review by Hisoka-X — APPROVED** [review]
LGTM. Thanks @corgy-w !

**Comment by corgy-w** [comment]
PTAL @wuchunfu

**Review by wuchunfu — APPROVED** [review]
LGTM, @corgy-w thanks

**Pull request merged**